tasks/push-disk-images/tests/mocks.sh

```diff
@@ -7,7 +7,7 @@ function internal-request() {
   echo Mock internal-request called with: $*
   echo $* >> $(workspaces.data.path)/mock_internal-request.txt
 
-  /home/utils/internal-request $@ -s false
+  /home/utils/internal-request "$@" -s false
```

Comment:
I don't think this is right. Maybe checkton told you to do this. But here it's actually intentional to call `internal-request` with the original parameters (and not one big parameter).

Reply:
actually, without this, some of my `-p` parameter values in the IRs were incorrectly quoted such that they were cut off.

Reply:
Hmm, interesting. I guess you're right and bash applies some special magic here for the special variable `$@`. Because normally I would expect the `""` around it to make just one argument out of it. But it seems it's not the case. Here are both variants and both work (but I don't have any arguments with spaces, so it's possible the quotes actually help you):
```
bash-5.2$ function outerfunc() { echo this is outerfunc params: $@; innerfunc $@; }
bash-5.2$ function innerfunc() { echo this is innerfunc params: $@; echo param1: $1; echo param2: $2; }
bash-5.2$ outerfunc param1 param2 param3
this is outerfunc params: param1 param2 param3
this is innerfunc params: param1 param2 param3
param1: param1
param2: param2
bash-5.2$ function outerfunc() { echo this is outerfunc params: $@; innerfunc "$@"; }
bash-5.2$ outerfunc param1 param2 param3
this is outerfunc params: param1 param2 param3
this is innerfunc params: param1 param2 param3
param1: param1
param2: param2
```

Anyway, as long as it works for you :)